### ISSUE TITLE
Graph.filterNodes is not available anymore

### ISSUE THREAD

**Issue by romainneutron** [cpettitt/graphlib#33]
Hello,

Previous versions of the graphlib provided a `Graph.filterNodes` method. It seems there is not such method anymore. Is it on purpose? Is there an easy way to achieve this?

Thanks for your work.

**Comment by ilyakharlamov** [comment]
```javascript
// tested with 1.0.1
graphlib.Graph.prototype.eachNode = function(func) {
  for (var id in this._nodes) {
    func(id, this._nodes[id]);
  }
};

graphlib.Graph.prototype.filterNodes = function(filter) {
  var copy = new this.constructor();
  copy.graph(this.graph());
  this.eachNode(function(u, value) {
    if (filter(u)) {
      copy.setNode(u, value);
    }
  });
  this.eachEdge(function(id, v, w, value) {
    if (copy.hasNode(v) && copy.hasNode(w)) {
      copy.setEdge(v, w, value);
    }
  });
  return copy;
};
graphlib.Graph.prototype.eachEdge = function(func) {
  var id, edge;
  for (id in this._edgeObjs) {
    edge=this._edgeObjs[id];
    func(id, edge.v, edge.w, this._edgeLabels[id]);
  }
};
```

**Comment by cpettitt** [comment]
Fixed in graphlib v1.0.6.
